src/entrypoints/dead-clicks-autocapture.ts

```diff
@@ -0,0 +1,209 @@
+const MUTATION_THRESHOLD_MS = 2500
+
+interface Click {
+    node: Element
+    timestamp: number
+    scrollDelayMs?: number
+    mutationDelayMs?: number
+    // if neither scroll nor mutation seen before threshold passed
+    absoluteDelayMs?: number
+}
+
+function asClick(event: Event): Click | null {
+    const eventTarget = getEventTarget(event)
+    if (eventTarget) {
+        return {
+            node: eventTarget,
+            timestamp: Date.now(),
+        }
+    }
+    return null
+}
+
+class _LazyLoadedDeadClicksAutocapture implements LazyLoadedDeadClicksAutocapture {
+    private _mutationObserver: MutationObserver | undefined
+    private _lastMutation: number | undefined
+    private _clicks: Click[] = []
+    private _checkClickTimer: number | undefined
+
+    constructor(readonly instance: PostHog) {}
+
+    start(observerTarget: Node) {
+        this._startClickObserver()
+        this._startScrollObserver()
+        this._startMutationObserver(observerTarget)
+    }
+
+    private _startMutationObserver(observerTarget: Node) {
+        if (!this._mutationObserver) {
+            this._mutationObserver = new MutationObserver((mutations) => {
+                this.onMutation(mutations)
+            })
+            this._mutationObserver.observe(observerTarget, {
+                attributes: true,
+                characterData: true,
+                childList: true,
+                subtree: true,
+            })
+        }
+    }
+
+    stop() {
+        this._mutationObserver?.disconnect()
+        this._mutationObserver = undefined
+        assignableWindow.removeEventListener('click', this._onClick)
+        assignableWindow.removeEventListener('scroll', this._onScroll, true)
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    private onMutation(_mutations: MutationRecord[]): void {
+        // we don't actually care about the content of the mutations, right now
+        this._lastMutation = Date.now()
+    }
+
+    private _startClickObserver() {
+        assignableWindow.addEventListener('click', this._onClick)
+    }
+
+    private _onClick = (event: Event): void => {
+        const click = asClick(event)
+        if (!isNull(click) && !this._ignoreClick(click)) {
+            this._clicks.push(click)
+        }
+
+        if (this._clicks.length && isUndefined(this._checkClickTimer)) {
+            this._checkClickTimer = assignableWindow.setTimeout(() => {
+                this._checkClicks()
+            }, 1000)
+        }
+    }
+
+    private _startScrollObserver() {
+        // setting the third argument to `true` means that we will receive scroll events for other scrollable elements
+        // on the page, not just the window
+        // see https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#usecapture
+        assignableWindow.addEventListener('scroll', this._onScroll, true)
```

Comment:
I _think_ we don't need to - we already track scrolls this way elsewhere
(but I'll double-check!)

Reply:
> You don't need to worry about the value of passive for the basic [scroll](https://developer.mozilla.org/en-US/docs/Web/API/Element/scroll_event) event. Since it can't be canceled, event listeners can't block page rendering anyway.

no, we're good here... although we should check through if any of the event listeners in posthog-js would benefit from being passive 👍